packages/next-swc/crates/core/src/next_ssg.rs

```diff
@@ -472,23 +498,36 @@ impl Fold for NextSsg {
                 | ExportSpecifier::Named(ExportNamedSpecifier {
                     exported: Some(exported),
                     ..
-                }) => !self.state.is_data_identifier(&exported),
-                ExportSpecifier::Named(s) => !self.state.is_data_identifier(&s.orig),
+                }) => self
+                    .state
+                    .is_data_identifier(&exported)
+                    .map(|is_data_identifier| !is_data_identifier),
+                ExportSpecifier::Named(s) => self
+                    .state
+                    .is_data_identifier(&s.orig)
+                    .map(|is_data_identifier| !is_data_identifier),
             };
 
-            if !preserve {
-                tracing::trace!("Dropping a export specifier because it's a data identifier");
+            match preserve {
+                Ok(should_preserve) => {
```

Comment:
`Ok(false)` seems better than pattern matching with an if expression.

You may need to add `Ok(_) => {}` branch.

Reply:
Thanks, new commit up.